src/server/context.ts

```diff
@@ -187,6 +187,11 @@ export class ServerContext {
       connInfo: ServeHandlerInfo = DEFAULT_CONN_INFO,
     ) {
       const url = new URL(req.url);
+      // Syntactically having double slashes in the pathname is valid per
+      // spec, but there is no behavior defined for that. Practically all
+      // servers normalize the pathname of a URL to not include double
+      // forward slashes.
+      url.pathname = url.pathname.replaceAll(/\/+/g, "/");
```

Comment:
Can you also check further down whether the original pathname matches the changed pathname? Because if it doesnt (there were double slashes), we should redirect to the normalized variant. Otherwise there will be a weird edge case where this logic matches as if there are no double slashes, but when a user does `new URL(req.url)`, they still get the "old" non-normalized URL.